src/VisualStudio/Core/Def/Implementation/TodoComments/IVisualStudioTodoCommentsService.cs

```diff
@@ -19,5 +22,12 @@ internal interface IVisualStudioTodoCommentsService
         /// analysis of the workspace to find todo comments
         /// </summary>
         void Start(CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Legacy entry-point to allow existing in-process languages to report todo comments.  These languages are
+        /// responsible for determining when to compute todo comments (for example, on <see
+        /// cref="Workspace.WorkspaceChanged"/>).  This can be called on any thread.
+        /// </summary>
+        Task ReportTodoCommentsAsync(Document document, ImmutableArray<TodoComment> todoComments, CancellationToken cancellationToken);
```

Comment:
obsolete

or instead should this be defined in the TS external access layer and implemented by grabbing the VisualStudioTodoCommentsService?

Reply:
made obsolete.  not sure if TS has EA.

Reply:
Yeah we do - for example, inline rename just got moved

Reply:
Great.  switched to EA pattern.